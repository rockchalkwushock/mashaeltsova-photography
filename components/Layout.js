import Meta from './Meta'

/**
 * REVIEW: Here is where I need to setup the grid.
 */

export default ({ children }) => (
  <main>
    {children}
    <Meta />
    <style jsx>{`
      main {
      }
    `}</style>
  </main>
)
