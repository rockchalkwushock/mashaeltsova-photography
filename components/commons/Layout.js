import { Component } from 'react'
import { initGA, logPageView } from '../../lib'

import Container from './Container'
import Footer from './Footer'
import Header from './Header'
import Icon from './Icon'
import Meta from './Meta'
import Photo from './Photo'

class Layout extends Component {
  componentDidMount() {
    if (!window.GA_INITIALIZED) {
      initGA()
      window.GA_INITIALIZED = true
    }
    logPageView()
  }
  render() {
    return (
      <div className="wrapper">
        <Meta />
        <Header>
          <Photo alt="Masha Eltsova Photography Logo" publicId="watermark" />
        </Header>
        {this.props.children}
        <Footer>
          <Container className="social">
            <Icon
              icon="facebook"
              size="2x"
              url="https://www.facebook.com/masha.eltsova"
            />
            <Icon
              icon="instagram"
              size="2x"
              url="https://www.instagram.com/mashaeltcovaphotography"
            />
            <Icon icon="vk" size="2x" url="https://www.vk.com/club65938200" />
          </Container>
          <Container className="copyright">
            <Icon
              icon="copyright"
              size="lg"
              url="https://rcws-development.com"
            />
          </Container>
        </Footer>
        <style jsx>{`
          .wrapper {
            display: flex;
            flex-direction: column;
            min-height: 100vh;
            text-align: center;
          }
        `}</style>
      </div>
    )
  }
}

export default Layout
