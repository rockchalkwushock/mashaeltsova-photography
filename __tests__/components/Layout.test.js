import React from 'react'
import { render } from 'enzyme'

import Layout from '../../components/commons/Layout'

describe('Component: Layout', () => {
  test('Renders without exploding', () => {
    const tree = render(<Layout />)
    expect(tree).toMatchSnapshot()
  })
})
