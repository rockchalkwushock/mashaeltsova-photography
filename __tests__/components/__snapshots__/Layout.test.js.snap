// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Layout Renders without exploding 1`] = `
<div
  class="jsx-3126043951 wrapper"
>
  <div
    class="jsx-3361249103"
  />
  <header
    class="jsx-3791225326"
  >
    <picture>
      <source
        media="(min-width: 1024px)"
        srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_1024/watermark"
      />
      <source
        media="(min-width: 768px)"
        srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_768/watermark"
      />
      <source
        media="(min-width: 540px)"
        srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_540/watermark"
      />
      <source
        media="(min-width: 425px)"
        srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_425/watermark"
      />
      <img
        alt="Masha Eltsova Photography Logo"
        src="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,dpr_auto,f_auto,o_90,q_auto:best,w_320/watermark"
      />
    </picture>
  </header>
  <footer
    class="jsx-3846709230"
  >
    <div
      class="jsx-1617167279 social"
    >
      <a
        href="https://www.facebook.com/masha.eltsova"
      >
        <span
          aria-hidden="true"
          class="fa fa-facebook fa-2x"
        />
      </a>
      <a
        href="https://www.instagram.com/mashaeltcovaphotography"
      >
        <span
          aria-hidden="true"
          class="fa fa-instagram fa-2x"
        />
      </a>
      <a
        href="https://www.vk.com/club65938200"
      >
        <span
          aria-hidden="true"
          class="fa fa-vk fa-2x"
        />
      </a>
    </div>
    <div
      class="jsx-1617167279 copyright"
    >
      <a
        href="https://rcws-development.com"
      >
        <span>
          <span
            aria-hidden="true"
            class="fa fa-copyright fa-lg"
          />
           2017 RCWS Development
        </span>
      </a>
    </div>
  </footer>
</div>
`;
